packages/office-ui-fabric-react/src/components/Link/Link.tsx

```diff
@@ -1,71 +1,9 @@
-import * as React from 'react';
-import {
-  BaseComponent,
-  anchorProperties,
-  autobind,
-  buttonProperties,
-  css,
-  getNativeProps
-} from '../../Utilities';
-import { ILink, ILinkProps } from './Link.types';
-import * as stylesImport from './Link.scss';
-const styles: any = stylesImport;
-
-export class Link extends BaseComponent<ILinkProps, any> implements ILink {
-  private _link: HTMLElement;
-
-  public render() {
-    let { disabled, children, className, href } = this.props;
-
-    return (
-      href ? (
-        <a
-          { ...getNativeProps(this.props, anchorProperties) }
-          className={ css(
-            'ms-Link',
-            styles.root,
-            className,
-            disabled && ('is-disabled ' + styles.isDisabled),
-            !disabled && styles.isEnabled
-          ) }
-          onClick={ this._onClick }
-          ref={ this._resolveRef('_link') }
-          target={ this.props.target }
-          aria-disabled={ disabled }
-        >
-          { children }
-        </a>
-      ) : (
-          <button
-            { ...getNativeProps(this.props, buttonProperties) }
-            className={ css(
-              'ms-Link',
-              styles.root,
-              className,
-              disabled && ('is-disabled ' + styles.isDisabled),
-              !disabled && styles.isEnabled
-            ) }
-            onClick={ this._onClick }
-            ref={ this._resolveRef('_link') }
-            aria-disabled={ disabled }
-          >
-            { children }
-          </button>
-        ));
-  }
-
-  public focus() {
-    if (this._link) {
-      this._link.focus();
-    }
-  }
-
-  @autobind
-  private _onClick(ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>) {
-    let { onClick } = this.props;
-
-    if (onClick) {
-      onClick(ev);
-    }
-  }
-}
+import { styled } from '../../Utilities';
+import { LinkBase } from './Link.base';
+import { ILinkProps } from './Link.types';
+import { getStyles } from './Link.styles';
+
+export const Link = styled(
```

Comment:
I would follow the trend of defining the types for `styled()`. IE:
```
export const Link = styled<ILinkProps, ILinkStyleProps, ILinkStyles>()
```

Reply:
Good call, I'll add those in.